lib/uinta/formatter/datadog.ex

```diff
@@ -0,0 +1,72 @@
+defmodule Uinta.Formatter.Datadog do
+  @moduledoc """
+
+  ## Installation
+
+  ```
+  config :logger, :console, format: {Uinta.Formatter.Datadog, :format}
+  ```
+
+  ## Features
+
+  ### Trace and Span id correlation
+
+  Takes trace ids and span ids in the OpenTelemetry format (as hex) and convert them into Datadog
+  format in the correct key. Logger.metadata must include the keys `:trace_id` and `:span_id`.
+  Be sure to add those two keys onto the allowlist in the `:logger` config. See
+  [Datadog's documentation](https://docs.datadoghq.com/tracing/connect_logs_and_traces/opentelemetry/)
+  for more details.
+
+  #### Example
+
+  ```elixir
+  ctx = OpenTelemetry.Tracer.current_span_ctx()
+  Logger.metadata([
+    trace_id: OpenTelemetry.Span.hex_trace_id(ctx),
+    span_id: OpenTelemetry.Span.hex_span_id(ctx)
+  ])
+  ```
+  """
+
+  alias Uinta.Formatter.Util
+  alias Uinta.Types
+
+  @doc """
+  See Uinta.formatter.format/4
+  """
+  @spec format(Types.level(), iodata(), Types.time(), Keyword.t()) :: iodata()
+  def format(level, message, timestamp, metadata) do
+    Util.format(level, message, timestamp, metadata)
+    |> add_datadog_trace(metadata)
```

Comment:
Do we need to add a new formatter for this? It seems like maybe we just separate the build/encode steps so folks can add their own metadata?